src/vcpkg/base/json.cpp

```diff
@@ -253,7 +253,7 @@ namespace vcpkg::Json
     }
     Value Value::number(double d) noexcept
     {
-        vcpkg::Checks::check_exit(VCPKG_LINE_INFO, isfinite(d));
+        vcpkg::Checks::check_exit(VCPKG_LINE_INFO, std::isfinite(d));
```

Comment:
this file (and the vcpkg-test one) should probably just include `<math.h>`

Reply:
For whatever reason, including `<math.h>` is not sufficient:

```
/home/bsiegert/src/vcpkg-tool/src/vcpkg/base/json.cpp: In static member function ‘static vcpkg::Json::Value vcpkg::Json::Value::number(double)’:
/home/bsiegert/src/vcpkg-tool/src/vcpkg/base/json.cpp:257:52: error: ‘isfinite’ was not declared in this scope
         vcpkg::Checks::check_exit(VCPKG_LINE_INFO, isfinite(d));
                                                    ^~~~~~~~
/home/bsiegert/src/vcpkg-tool/src/vcpkg/base/json.cpp:257:52: note: suggested alternatives:
In file included from /usr/include/g++/random:38:0,
                 from /home/bsiegert/src/vcpkg-tool/include/pch.h:37,
                 from <command-line>:0:
/usr/include/g++/cmath:585:5: note:   ‘std::isfinite’
     isfinite(_Tp __x)
     ^~~~~~~~
In file included from /home/bsiegert/src/vcpkg-tool/include/vcpkg/base/format.h:14:0,
                 from /home/bsiegert/src/vcpkg-tool/include/vcpkg/base/messages.h:7,
                 from /home/bsiegert/src/vcpkg-tool/include/vcpkg/base/basic_checks.h:4,
                 from /home/bsiegert/src/vcpkg-tool/include/vcpkg/base/checks.h:3,
                 from /home/bsiegert/src/vcpkg-tool/include/vcpkg/base/files.h:5,
                 from /home/bsiegert/src/vcpkg-tool/src/vcpkg/base/json.cpp:1:
/home/bsiegert/src/vcpkg-tool/_deps/fmt-src/include/fmt/format.h:1957:22: note:   ‘fmt::v8::detail::isfinite’
 FMT_CONSTEXPR20 bool isfinite(T value) {
                      ^~~~~~~~
```

I think this may be due to the way `isfinite` is declared as a macro:

```c
#define isfinite(__x)   __fpmacro_unary_floating(isfinite, __x)
```

Reply:
Note: 
`std::isfinite` is a free function from C++11  header `cmath`. 
https://en.cppreference.com/w/cpp/numeric/math/isfinite 
`isfinite` is a macro from C99 header `math.h`. 
https://en.cppreference.com/w/c/numeric/math/isfinite 

Does the macro need extra care, such wrapping in `( ... )`?

Reply:
> Every C header other than [`<complex.h>`], [`<iso646.h>`], [`<stdalign.h>`],
[`<stdatomic.h>`], [`<stdbool.h>`], and [`<tgmath.h>`], each of which has a name of the form `<name.h>`, behaves as if each name placed in the standard library namespace by the corresponding `<cname>` header is placed within the global namespace scope, except for the functions described in [\[sf.cmath\]], the declaration of `std​::​byte` ([\[cstddef.syn\]]), and the functions and function templates described in [\[support.types.byteops\]]. It is unspecified whether these names are first declared or defined within namespace scope ([\[basic.scope.namespace\]]) of the namespace std and are then injected into the global namespace scope by explicit [using-declaration]s ([\[namespace.udecl\]]).

[`<complex.h>`]: http://eel.is/c++draft/support.c.headers#header:%3ccomplex.h%3e
[`<iso646.h>`]: http://eel.is/c++draft/support.c.headers#header:%3ciso646.h%3e
[`<stdalign.h>`]: http://eel.is/c++draft/support.c.headers#header:%3cstdalign.h%3e
[`<stdatomic.h>`]: http://eel.is/c++draft/stdatomic.h.syn#header:%3cstdatomic.h%3e
[`<stdbool.h>`]: http://eel.is/c++draft/support.c.headers#header:%3cstdbool.h%3e
[`<tgmath.h>`]: http://eel.is/c++draft/support.c.headers#header:%3ctgmath.h%3e
[\[sf.cmath\]]: http://eel.is/c++draft/sf.cmath
[\[cstddef.syn\]]: http://eel.is/c++draft/cstddef.syn
[\[support.types.byteops\]]: http://eel.is/c++draft/support.types.byteops
[\[basic.scope.namespace\]]: http://eel.is/c++draft/basic.scope.namespace
[using-declaration]: http://eel.is/c++draft/namespace.udecl#nt:using-declaration
[\[namespace.udecl\]]: http://eel.is/c++draft/namespace.udecl

Reply:
It seems like @bsiegert's implementation may not be conforming, though...

Reply:
I still suspect we have a `<cmath>` floating around though? After all, this compiled and it shouldn't have without one.

Reply:
From what I can tell, `<cmath>` is included by `format.h`, which is part of the fmt dependency. format.h also includes its own definition of `isfinite`. I have no idea why `isfinite` would not be declared in the scope then.

Reply:
@bsiegert can you try adding a `<math.h>` include without the other changes? I can't personally test on BSD

Reply:
(If that does not fix the problem, I think we need to pursue a more complete change of always std:: ing all the things)

Reply:
I tried that, and it fails in the same way. 

FWIW, there is an unconditional `<cmath>` include from `<fmt/format.h>` as provided by the libfmt dependency.